### PR TITLE
docs(store): add guide for provideMockStore

### PR DESCRIPTION
**Pull request by lacolaco**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

There is no guide for `provideMockStore` API.

## What is the new behavior?

Added

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

maybe relates to https://github.com/ngrx/platform/issues/1383

### CODE REVIEW

File: projects/ngrx.io/content/guide/store/testing.md [hunk 1]
```diff
@@ -142,3 +142,59 @@ describe('My Selectors', () => {
   });
 });
 </code-example>
+
+### provideMockStore
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/store/testing.md]:
```suggestion
### Using a Mock Store
```

File: projects/ngrx.io/content/guide/store/testing.md [hunk 2]
```diff
@@ -142,3 +142,59 @@ describe('My Selectors', () => {
...
+
+Provides a *mock store* for testing effects or something which has a dependency on `Store` without setting up reducers.
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/store/testing.md]:
```suggestion
The `provideMockStore()` method registers providers that allow you to mock out the `Store` for testing functionality that has a dependency on `Store` without setting up reducers. 
```

File: projects/ngrx.io/content/guide/store/testing.md [hunk 3]
```diff
@@ -142,3 +142,59 @@ describe('My Selectors', () => {
   });
 });
 </code-example>
+
+### provideMockStore
+
+Provides a *mock store* for testing effects or something which has a dependency on `Store` without setting up reducers. 
+You can write tests validating behaviors corresponding to the specific state snapshot easily.
+
+Note: All dispatched actions don't affect to the state, but you can see them in the `Actions` stream.
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/store/testing.md]:
```suggestion
<div class="alert is-helpful">

**Note:** All dispatched actions don't affect to the state, but you can see them in the `Actions` stream.

</div>
```

File: projects/ngrx.io/content/guide/store/testing.md [hunk 4]
```diff
@@ -142,3 +142,59 @@ describe('My Selectors', () => {
...
+
+Usage: 
+
+<code-example header="auth.guard.spec.ts">
+import { TestBed } from '@angular/core/testing';
+import { provideMockActions } from '@ngrx/effects/testing';
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/store/testing.md]: Remove unused imports

File: projects/ngrx.io/content/guide/store/testing.md [hunk 5]
```diff
@@ -142,3 +142,59 @@ describe('My Selectors', () => {
...
+import { Store } from '@ngrx/store';
+import { provideMockStore } from '@ngrx/store/testing';
+import { cold } from 'jasmine-marbles';
+
+import { AuthGuard } from '../guards/auth.guard';
+import * as AuthActions from '../actions/auth-actions';
+
+describe('Auth Guard', () => {
+  let guard: AuthGuard;
+  let mockStore: MockStore<{ loggedIn: boolean } >;
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/store/testing.md]:
```suggestion
  let store: MockStore<{ loggedIn: boolean } >;
```

File: projects/ngrx.io/content/guide/store/testing.md [hunk 6]
```diff
@@ -142,3 +142,59 @@ describe('My Selectors', () => {
...
+      ],
+      providers: [
+        AuthGuard,
+        provideMockStore({ initialState }),
+        // other providers
+      ],
+    });
+
+    guard = TestBed.get(AuthGuard);
+    mockStore = TestBed.get(Store);
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/store/testing.md]:
```suggestion
    store = TestBed.get(Store);
```

File: projects/ngrx.io/content/guide/store/testing.md [hunk 7]
```diff
@@ -142,3 +142,59 @@ describe('My Selectors', () => {
...
+  });
+
+  it('should return false if the user state is not logged in', () => {
+    const expected = cold('(a|)', { a: false });
+
+    expect(guard.canActivate()).toBeObservable(expected);
+  });
+
+  it('should return true if the user state is logged in', () => {
+    mockStore.setState({ loggedIn: true });
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/store/testing.md]:
```suggestion
    store.setState({ loggedIn: true });
```

File: projects/ngrx.io/content/guide/store/testing.md (hunk 5 above)
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/store/testing.md]: The greater than/less than arrows need to be HTML entities to display correctly

**Comment by brandonroberts** [comment]
@lacolaco I left some feedback. Will you move this to the top and change the title for `Provide Store for testing` to `Using Store for Integration Testing`.

Thanks!

File: projects/ngrx.io/content/guide/store/testing.md [hunk 8]
```diff
@@ -1,6 +1,65 @@
 # Testing
 
-### Providing Store for testing
+### Using a Mock Store
+
+The `provideMockStore()` method registers providers that allow you to mock out the `Store` for testing functionality that has a dependency on `Store` without setting up reducers.
```
Comment (lacolaco) [inline projects/ngrx.io/content/guide/store/testing.md]: I think this is not a `method` but just `function`. isn't?

Comment (brandonroberts) [reply projects/ngrx.io/content/guide/store/testing.md]: Yep, `function` is better.

**Comment by lacolaco** [comment]
@brandonroberts sorry for noisy commits! it's done!

**Comment by swapnil0545** [comment]
@brandonroberts 
I have a naive question: I dont see this https://previews.ngrx.io/pr1591-3224212/guide/store/testing
 on the real ngrx site. How do we come to know when the site gets updated?

**Comment by brandonroberts** [comment]
@swapnil0545 It needs to be deployed, which I'll do today. You can also see what's latest on master if you go to https://next.ngrx.io

**Pull request merged**